Program.cs

```diff
@@ -88,6 +110,34 @@ internal static async Task<int> Main(string[] args)
 
                 var manifest = (Manifest?) serializer.Deserialize(reader) ?? throw new InvalidDataException();
 
+                bool matching = false;
+                for (int i = 0; i < compareManifests.Count; i++)
+                {
+                    var xmlNode = compareManifests[i] ?? throw new InvalidDataException();
+                    var stream = new MemoryStream();
+                    var writer = new StreamWriter(stream);
+                    await writer.WriteAsync(xmlNode.OuterXml);
+                    await writer.FlushAsync();
+
+                    stream.Position = 0;
+                    
+                    var compareManifest = (Manifest?) serializer.Deserialize(stream) ?? throw new InvalidDataException();
+                    
+                    if (compareManifest.Name != manifest.Name) continue;
+                    
+                    if (compareManifest.Name == manifest.Name &&

```

Comment:
This should just be a .Equals/== implementing for Manifest. You can have it be a record to get == from Equals for brevity too if desired. I'd also just pattern match out the nulls here. Also note that atm if the first is non-null and the second is null it'll explode. 

```cs
if ((compareManifest.Dependencies is not cDeps {} and manifests.Dependencies is not deps {}) or cDeps.SequenceEquals(deps))
```